### PR TITLE
Add RememberMeDsl

### PR DESCRIPTION
**Pull request by IvanPavlov1995**
Issue: gh-9319

### CODE REVIEW

**Review by eleftherias — CHANGES_REQUESTED** [review]
Thanks for the PR @IvanPavlov1995! 
I left some comments inline.

File: config/src/main/kotlin/org/springframework/security/config/web/servlet/RememberMeDsl.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ * @property useSecureCookie whether the cookie should be flagged as secure or not
+ * @property alwaysRemember whether the cookie should always be created even if
+ * the remember-me parameter is not set. Defaults to `false`
+ */
+@SecurityMarker
+class RememberMeDsl {
+    var authenticationSuccessHandler: AuthenticationSuccessHandler? = null
+    var key: String? = null
+    var rememberMeServices: RememberMeServices? = null
+    var rememberMeParameter: String = DEFAULT_REMEMBER_ME_NAME
```
[the remark this answers is not in the document]
Comment (IvanPavlov1995) [reply config/src/main/kotlin/org/springframework/security/config/web/servlet/RememberMeDsl.kt]: Done

File: config/src/test/kotlin/org/springframework/security/config/web/servlet/RememberMeDslTest.kt [hunk 2]
```diff
@@ -0,0 +1,564 @@
+/*
```
[the remark this answers is not in the document]
Comment (IvanPavlov1995) [reply config/src/test/kotlin/org/springframework/security/config/web/servlet/RememberMeDslTest.kt]: Thanks for pointing that, fixed

File: config/src/main/kotlin/org/springframework/security/config/web/servlet/RememberMeDsl.kt [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    var tokenValiditySeconds: Int? = null
+    var useSecureCookie: Boolean? = null
+    var alwaysRemember: Boolean = false
+
+    internal fun get(): (RememberMeConfigurer<HttpSecurity>) -> Unit {
+        return { rememberMe ->
+            authenticationSuccessHandler?.also { rememberMe.authenticationSuccessHandler(authenticationSuccessHandler) }
+            key?.also { rememberMe.key(key) }
+            rememberMeServices?.also { rememberMe.rememberMeServices(rememberMeServices) }
+            rememberMe.rememberMeParameter(rememberMeParameter)
```
[the remark this answers is not in the document]
Comment (IvanPavlov1995) [reply config/src/main/kotlin/org/springframework/security/config/web/servlet/RememberMeDsl.kt]: Done.
I have a question regarding `also` syntax. Is there a reason not to use function reference in such configurations? I mean following is shorter and with that I don't have to use `!!` operator:
```kotlin
useSecureCookie?.also(rememberMe::useSecureCookie)
```

[the remark this answers is not in the document]
Comment (eleftherias) [reply config/src/main/kotlin/org/springframework/security/config/web/servlet/RememberMeDsl.kt]: There is no particular reason. We could use function references instead. I will merge this PR as is, but feel free to create an additional PR if you think it's something we should change.

**Pull request merged**